### ISSUE TITLE
Incorrect results from 'nthRoot'

### ISSUE THREAD

**Issue by FSMaxB** [josdejong/mathjs#397]
There are numerical instabilities in the `nthRoot` function, see the following example:

```js
$ ./repl.js 
> math.eval('sqrt(2^1000)/2^500');
1
> math.eval('nthRoot(2^1000,2)/2^500');
2.5822498780869086e+120
> 
```

I haven't looked at the code of `nthRoot` or at possible algorithms yet.

Are there any algorithms that handle those instabilities better?

**Comment by josdejong** [comment]
Thanks! Turns out there is an infinite loop protection with a too low maximum count ([here in the code](https://github.com/josdejong/mathjs/blob/develop/lib/function/arithmetic/nthRoot.js#L236)). `iMax` is 100, but in your example there are 506 iterations needed. We can at least increase `iMax` to say 1000 or 10000. I'm not sure whether it would be safe to just remove the loop protection.

**Comment by FSMaxB** [comment]
Under what conditions would this be an infinite loop? If I understand it correctly, this comes down to the question if there are parameters for which the algorithm doesn't converge (not mathematically but numerically ).

If it's not sure if this converges, `iMax` should be set to something that doesn't take too long on slow hardware (let's say 10-20 seconds).

Another suggestion: Throw an error when `iMax` is reached and provide an optional third parameter for the number of iterations.

I think an error should be thrown because otherwise the result is incorrect and you have no way of knowing that it is.

**Comment by josdejong** [comment]
I've just pushed a fix

**Comment by FSMaxB** [comment]
Thanks.

Should this issue be closed now or after the release of v2?

**Comment by josdejong** [comment]
I've changed the max iterations to 10000 (it's typically between 5 and 20, and for very small or large values it can be in the order of 500). 

The function now throws an error when the maximum iterations is reached. However, so far I couldn't come up with a case where this happens. I think adding an optional third parameter will be a bit overkill, the method should just work.

I also saw that the function didn't work at all for small values ( < 1e-16 ) due to the epsilon. I've fixed that too by comparing x against xPrev in the loop.

The issue can stay open until the issue is applied in the first next release (will automatically be closed by github then). We could also do a bugfix for the v1 branch.

**Comment by FSMaxB** [comment]
Is throwing an error API-Breaking?

**Comment by josdejong** [comment]
I would say not, it's fixing a bug.

**Comment by FSMaxB** [comment]
I've ported this fix to `v1_develop`.

**Comment by josdejong** [comment]
Thanks Max, I will do a bugfix release this weekend (just back from vacation - it's time to get this `v2` finally finished).

**Comment by josdejong** [comment]
I've released v1.7.1 including this bug fix.
